### ISSUE TITLE
undici installed in a TypeScript project errors with `...types/client.d.ts:5:33 - error TS2307: Cannot find module 'abort-controller' or its corresponding type declarations.`

### ISSUE THREAD

**Issue by Jolg42** [nodejs/undici#480]
When updgrading from 2.1.1 to [2.2.0](https://github.com/nodejs/undici/releases/tag/v2.2.0) I got this error in TypeScript:
```
> tsc -d

../../node_modules/.pnpm/undici@2.2.0/node_modules/undici/types/client.d.ts:5:33 - error TS2307: Cannot find module 'abort-controller' or its corresponding type declarations.

5 import { AbortController } from 'abort-controller'
                                  ~~~~~~~~~~~~~~~~~~

Found 1 error.
```

The source should be this line https://github.com/nodejs/undici/blob/master/types/client.d.ts#L5

So here the `abort-controller` package is missing and because TypeScript cannot find it, it errors.
Meaning that it needs to be installed / be a required dependency.

I added the package as a dev dependency in my project and it worked 😃 

Maybe this would need to be documented or the dependency required? (Note: I am not using abort-controller functionality here).

Note: Many Thanks for the types! 💚

**Comment by ronag** [comment]
@Ethan-Arrowood

**Comment by Ethan-Arrowood** [comment]
abort-controller should be shipped in `@types/node` now. 

@Jolg42 do you have node types installed? If so which version?

**Comment by Ethan-Arrowood** [comment]
I noticed that in the [aborting a request](https://github.com/nodejs/undici#aborting-a-request) section

> To use AbortController you must `npm i abort-controller`

So yeah i don't actually think anything is wrong here. We might need to make a new section of the docs that calls this out better though. Like maybe a notice at the top a long the lines of "If you want to use AbortController you must install it yourself with `npm i abort-controller`.

In regards to types though, I realize that this might still error out since that type will be missing. I need to investigate a bit further on how to remove this type dependency (since its not a dependency of the project), and then patch the type ourselves with something like `unknown`. I envision there is some decent solution here.

**Comment by Ethan-Arrowood** [comment]
I'm having a rough morning 😂

Okay so solution:

Node 15 ships abort-controller. But types haven't been published for @types/node@15.0.0

Node 14 doesn't and requires a shim `abort-controller`. Which ships its own types.

Since the definition for AbortController is quite small, im going to copy and paste it into Undici source with a comment block that links back here + explains that it'll be removed when it is apart of Node LTS.

I can't think of a better solution at this time, but welcome to other ideas.

**Comment by Jolg42** [comment]
Sounds good to me 💯 Thanks!
